include/mra/misc/convolutiondata.h

```diff
@@ -0,0 +1,105 @@
+#ifndef MRA_CONVOLUTIONDATA_H
+#define MRA_CONVOLUTIONDATA_H
+
+#include "mra/ops/inner.h"
+#include "mra/misc/hash.h"
+#include "mra/misc/misc.h"
+#include "mra/misc/types.h"
+#include "mra/misc/adquad.h"
+#include "mra/misc/platform.h"
+#include "mra/misc/autocorr.h"
+#include "mra/tensor/tensorview.h"
+
+
+namespace mra {
+
+  template <typename T, Dimension NDIM>
+  class ConvolutionData {
+
+    private:
+      size_type K;
+      Level n;
+      int npt;                  // number of quadrature points
+      Translation lx;
+      Tensor<T, 1> quad_x;      // quadrature points
+      Tensor<T, 1> quad_w;      // quadrature weights
+      Tensor<T, 3> autocorr;    // autocorrelation coefficients
+      Tensor<T, 3> c;           // autocorrelation coefficients
+      Tensor<T, 2> rnlij;       // rnlij coefficients
+      Tensor<T, 1> coeff;       // coefficients for the convolution
+      Tensor<T, 1> rnlp;        // rnlp coefficients
+
+
+      void autoc(){
+        detail::autocorr_get(K, autocorr.data());
+
+        auto c_view = c.current_view();
+        c_view(Slice(0, K-1), Slice(0, K-1), Slice(0, 2*K-1)) = autocorr(Slice(0, K-1), Slice(0, K-1), Slice(0, 2*K-1));
+        c_view(Slice(0, K-1), Slice(0, K-1), Slice(2K, 4*K-1)) = autocorr(Slice(0, K-1), Slice(0, K-1), Slice(2*K, 4*K-1));
+      }
+
+      void rnlij(const Level n, const Translation lx){
+        Tensor<T, 1> R(4*K);
+        R_view = R.current_view();
+        R_view(Slice(0, 2*K-1)) = rnlp(n, lx-1);
```

Comment:
Do you think  `lx-1` is a valid operation under our implementation?

Reply:
It should be:

```
using Translation = uint32_t;
```